compiler/passes/insertLineNumbers.cpp

```diff
@@ -77,356 +78,393 @@ int getFilenameLookupPosition(std::string name) {
+    // from the new corresponding field in the argument bundle.
+    DefExpr* bundleArg = toDefExpr(fn->formals.tail);
+    AggregateType* bundleType = toAggregateType(bundleArg->sym->typeInfo());
 
-      INT_ASSERT(fn);
-      INT_ASSERT(fn->substitutionsPostResolve.size() > 0);
+    VarSymbol* lineField = newTemp("_ln", dtInt[INT_SIZE_DEFAULT]);
+    bundleType->fields.insertAtTail(new DefExpr(lineField));
 
-      VarSymbol* var = toVarSymbol(fn->substitutionsPostResolve[0].value);
+    VarSymbol* lineLocal = newTemp("_ln", dtInt[INT_SIZE_DEFAULT]);
 
-      INT_ASSERT(var);
-      INT_ASSERT(var->immediate);
-      INT_ASSERT(var->immediate->const_kind == CONST_KIND_STRING);
+    fn->insertAtHead("'move'(%S, '.v'(%S, %S))", lineLocal, bundleArg->sym,
+                     lineField);
+    fn->insertAtHead(new DefExpr(lineLocal));
 
-      const char* cmdLineSetting = astr("<command line setting of '",
-                                        var->immediate->v_string.c_str(),
-                                        "'>");
+    // Same thing, just for the filename index now.
 
-      int         filenameIdx    = getFilenameLookupPosition(cmdLineSetting);
+    VarSymbol* fileField = newTemp("_fn", dtInt[INT_SIZE_32]);
+    bundleType->fields.insertAtTail(new DefExpr(fileField));
 
-      file = new_IntSymbol(filenameIdx, INT_SIZE_32);
+    VarSymbol* fileLocal = newTemp("_fn", dtInt[INT_SIZE_32]);
 
-    } else {
-      // Apply the line number from the call AST node
-      line = new_IntSymbol(call->linenum());
+    fn->insertAtHead("'move'(%S, '.v'(%S, %S))", fileLocal, bundleArg->sym,
+                     fileField);
+    fn->insertAtHead(new DefExpr(fileLocal));
 
-      int filenameIdx = getFilenameLookupPosition(call->fname());
+    result = {lineLocal, fileLocal};
 
-      file = new_IntSymbol(filenameIdx, INT_SIZE_32);
-    }
   } else {
-    // This branch handles the case in which we should be using
-    // file/line arguments in the enclosing function.
+    ArgSymbol* line =
+        new ArgSymbol(INTENT_CONST_IN, "_ln", dtInt[INT_SIZE_DEFAULT]);
+    fn->insertFormalAtTail(line);
 
-    // Create them if they don't exist yet.
-    if (lineArg == NULL)
-      lineArg = newLine(fn);
-    if (fileArg == NULL)
-      fileArg = newFile(fn);
+    ArgSymbol* file = new ArgSymbol(INTENT_IN, "_fn", dtInt[INT_SIZE_32]);
+    fn->insertFormalAtTail(file);
 
-    line = lineArg;
-    file = fileArg;
+    result = {line, file};
   }
 
-  if (call->isPrimitive(PRIM_GET_USER_FILE) ||
-      call->isPrimitive(PRIM_GET_USER_LINE)) {
+  lineAndFilenameMap.emplace_hint(it, fn, result);
 
-    // Replace these primitives with whatever line/file we chose
-    // (probably an argument)
-    if (call->isPrimitive(PRIM_GET_USER_FILE)) {
-      call->replace(new SymExpr(file));
-    } else if (call->isPrimitive(PRIM_GET_USER_LINE)) {
-      call->replace(new SymExpr(line));
-    }
+  for (CallExpr* call : *fn->calledBy) {
```

Comment:
Seems like you know that fn->calledBy is computed/current because you just ran ComputeCallSitesPass before insertLineNumbers?

Reply:
That is correct, are you wanting another comment here on that dependence?

Reply:
I think it would be nice, yes. I don't remember what would happen if you didn't run ComputeCallSites first. If it doesn't fail in an obvious manner, it might be nice to add some assertions or something as well.

Reply:
The pass previously ran `compute_call_sites` FYI but will add a comment